code/controllers/subsystems/transcore_vr.dm

```diff
@@ -37,49 +44,60 @@ SUBSYSTEM_DEF(transcore)
 
 /datum/controller/subsystem/transcore/proc/process_implants(resumed = 0)
 	if (!resumed)
-		src.current_run = implants.Copy()
+		// Create a flat list of every implant in every db with a value of the db they're in
+		src.current_run.Cut()
+		for(var/key in databases)
+			var/datum/transcore_db/db = databases[key]
+			for(var/obj/item/weapon/implant/backup/imp as anything in db.implants)
+				src.current_run[imp] = db
 
 	var/list/current_run = src.current_run
 	while(current_run.len)
 		var/obj/item/weapon/implant/backup/imp = current_run[current_run.len]
+		var/datum/transcore_db/db = current_run[imp]
 		current_run.len--
 
 		//Remove if not in a human anymore.
 		if(!imp || !isorgan(imp.loc))
-			implants -= imp
+			db.implants -= imp
 			continue
 
 		//We're in an organ, at least.
 		var/obj/item/organ/external/EO = imp.loc
 		var/mob/living/carbon/human/H = EO.owner
 		if(!H)
-			implants -= imp
+			db.implants -= imp
 			continue
 
 		//In a human
 		BITSET(H.hud_updateflag, BACKUP_HUD)
 
 		if(H == imp.imp_in && H.mind && H.stat < DEAD)
-			SStranscore.m_backup(H.mind,H.nif)
+			db.m_backup(H.mind,H.nif)
 			persist_nif_data(H)
 
 		if(MC_TICK_CHECK)
 			return
 
 /datum/controller/subsystem/transcore/proc/process_backups(resumed = 0)
 	if (!resumed)
-		src.current_run = backed_up.Copy()
+		// Create a flat list of every implant in every db with a value of the db they're in
+		src.current_run.Cut()
+		for(var/key in databases)
+			var/datum/transcore_db/db = databases[key]
+			for(var/datum/transhuman/mind_record/mr as anything in db.backed_up)
```

Comment:
Looking at code elsewhere, I think  `db.backed_up` contains names (string) not mind_records.   The mind_record is the associated value.

Reply:
Fixed